File: o-b/src/main/java/com/soft/ob/entity/Elderly.java
package com.soft.ob.entity;

import lombok.Data;
import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "elderly")
@Data
public class Elderly {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "uuid", unique = true, nullable = false)
    private String uuid; // 老人唯一标识
    
    @Column(name = "username", unique = true, nullable = false)
    private String username;
    
    @Column(name = "password", nullable = false)
    private String password;
    
    @Column(name = "salt")
    private String salt; // 密码盐值
    
    @Column(name = "name", nullable = false)
    private String name;
    
    @Column(name = "age")
    private Integer age;
    
    @Column(name = "gender")
    private String gender;
    
    @Column(name = "phone")
    private String phone;
    
    @Column(name = "id_number", unique = true)
    private String idNumber; // 身份证号
    
    @Column(name = "address")
    private String address;
    
    @Column(name = "avatar")
    private String avatar; // 头像
    
    @Column(name = "emergency_contact_name")
    private String emergencyContactName;
    
    @Column(name = "emergency_contact_phone")
    private String emergencyContactPhone;
    
    @Column(name = "health_status")
    private String healthStatus;
    
    @Column(name = "special_medical_needs")
    private String specialMedicalNeeds;
    
    @Column(name = "admission_date")
    private LocalDate admissionDate;
    
    @Column(name = "is_active")
    private Boolean isActive = true;
    
    @Column(name = "login_attempts")
    private Integer loginAttempts = 0;
    
    @Column(name = "locked_until")
    private LocalDateTime lockedUntil;
    
    @Column(name = "create_time")
    private LocalDateTime createTime;
    
    @Column(name = "update_time")
    private LocalDateTime updateTime;
    
    @Column(name = "last_login_time")
    private LocalDateTime lastLoginTime;
    
    @Column(nullable = false)
    private String roomNumber;
    
    @Column
    private LocalDate birthDate;
    
    @Column
    private String notes;
}

File: o-f/src/models/Elderly.js
export class Elderly {
  constructor({
    id = null,
    name = '',
    age = null,
    gender = '',
    phone = '',
    address = '',
    birthDate = null,
    emergencyContactName = '',
    emergencyContactPhone = '',
    healthStatus = '',
    specialMedicalNeeds = '',
    roomNumber = '',
    admissionDate = null,
    isActive = true,
    notes = ''
  } = {}) {
    this.id = id
    this.name = name
    this.age = age
    this.gender = gender
    this.phone = phone
    this.address = address
    this.birthDate = birthDate
    this.emergencyContactName = emergencyContactName
    this.emergencyContactPhone = emergencyContactPhone
    this.healthStatus = healthStatus
    this.specialMedicalNeeds = specialMedicalNeeds
    this.roomNumber = roomNumber
    this.admissionDate = admissionDate
    this.isActive = isActive
    this.notes = notes
  }

  // 静态方法：从API响应创建Elderly实例
  static fromAPI(data) {
    return new Elderly({
      id: data.id,
      name: data.name,
      age: data.age,
      gender: data.gender,
      phone: data.phone,
      address: data.address,
      birthDate: data.birthDate,
      emergencyContactName: data.emergencyContactName,
      emergencyContactPhone: data.emergencyContactPhone,
      healthStatus: data.healthStatus,
      specialMedicalNeeds: data.specialMedicalNeeds,
      roomNumber: data.roomNumber,
      admissionDate: data.admissionDate,
      isActive: data.isActive,
      notes: data.notes
    })
  }

  // 静态方法：从API响应数组创建Elderly实例数组
  static fromAPIList(dataList) {
    return dataList.map(item => Elderly.fromAPI(item))
  }

  // 转换为API请求格式
  toAPI() {
    return {
      id: this.id,
      name: this.name,
      age: this.age,
      gender: this.gender,
      phone: this.phone,
      address: this.address,
      birthDate: this.birthDate,
      emergencyContactName: this.emergencyContactName,
      emergencyContactPhone: this.emergencyContactPhone,
      healthStatus: this.healthStatus,
      specialMedicalNeeds: this.specialMedicalNeeds,
      roomNumber: this.roomNumber,
      admissionDate: this.admissionDate,
      isActive: this.isActive,
      notes: this.notes
    }
  }
}